pkg/kubectl/cmd/util/factory.go

```diff
@@ -337,25 +337,22 @@ func getSchemaAndValidate(c *client.RESTClient, data []byte, group, version stri
 }
 
 func (c *clientSwaggerSchema) ValidateBytes(data []byte) error {
-	version, _, err := runtime.UnstructuredJSONScheme.DataVersionAndKind(data)
+	version, kind, err := runtime.UnstructuredJSONScheme.DataVersionAndKind(data)
 	if err != nil {
 		return err
 	}
 	if ok := registered.IsRegisteredAPIVersion(version); !ok {
 		return fmt.Errorf("API version %q isn't supported, only supports API versions %q", version, registered.RegisteredVersions)
 	}
-	// First try stable api, if we can't validate using that, try experimental.
-	// If experimental fails, return error from stable api.
-	// TODO: Figure out which group to try once multiple group support is merged
-	//       instead of trying everything.
-	err = getSchemaAndValidate(c.c.RESTClient, data, "api", version)
-	if err != nil && c.ec != nil {
-		errExp := getSchemaAndValidate(c.ec.RESTClient, data, "experimental", version)
-		if errExp == nil {
-			return nil
+	resource, _ := meta.KindToResource(kind, false)
+	group, err := api.RESTMapper.GroupForResource(resource)
```

Comment:
Sorry, I've been busy with things. I have changed this to error in my local copy. I haven't written any integration tests yet.